python/prairielearn.py

```diff
@@ -52,6 +52,39 @@ class QuestionData(TypedDict):
 class ElementTestData(QuestionData):
     test_type: Literal['correct', 'incorrect', 'invalid']
 
+
+def set_weighted_score_data(data: QuestionData) -> None:
+    "Sets main question score to be weighted average of all partial scores"
+    # TODO change to check that all weights are unset if using an equal weighting
+    weight_total = 0
+    score_total = 0.0
+    for part in data["partial_scores"].values():
+        score = part["score"]
+        weight = part["weight"]
```

Comment:
There are questions that skip the weight in the `grade` function, it would probably make sense to default it to 1 for compatibility in these cases.
```suggestion
        weight = part.get("weight", 1)
```

Reply:
I think it makes more sense to have it default to 1 instead of a parameter. This is also done in other parts of the code, e.g.:
https://github.com/PrairieLearn/PrairieLearn/blob/8ced3f982a733fc3ff2c879cce65f3e222fbb511/sprocs/instance_questions_update_score.sql#L138
https://github.com/PrairieLearn/PrairieLearn/blob/8ced3f982a733fc3ff2c879cce65f3e222fbb511/elements/pl-integer-input/pl-integer-input.py#L10

Reply:
I think it's reasonable to be consistent then. Will change the parameter default to 1.